robot_description_kinematics:
  fr3_arm:
    kinematics_solver: lma_kinematics_plugin/LMAKinematicsPlugin
    kinematics_solver_ik_links:
      - fr3_link8
    kinematics_solver_search_resolution: 0.005
    kinematics_solver_timeout: 0.005
  fr3_manipulator:
    kinematics_solver: lma_kinematics_plugin/LMAKinematicsPlugin
    kinematics_solver_ik_links:
      - fr3_hand_tcp
    kinematics_solver_search_resolution: 0.005
    kinematics_solver_timeout: 0.005
  # hand:
  #   kinematics_solver: null
  #   kinematics_solver_ik_links: null
  #   kinematics_solver_search_resolution: null
  #   kinematics_solver_timeout: null
  arm:
    kinematics_solver: pick_ik/PickIkPlugin
    kinematics_solver_timeout: 0.05
    kinematics_solver_attempts: 3
    mode: local
    position_scale: 1.0
    rotation_scale: 0.5
    position_threshold: 0.001
    orientation_threshold: 0.01
    cost_threshold: 0.001
    minimal_displacement_weight: 0.0
    gd_step_size: 0.0001
